lib/perl/Genome/Model/Command/Admin/RemoveDiskAllocationsFromTestdb.pm

```diff
@@ -0,0 +1,348 @@
+
+        } else {
+            # This allocation was deleted in the test database?!
+            undef($next_tmpl_allocation);
+
+        }
+    }
+
+    $self->disconnect_database_handles();
+    return @new_allocations_in_database;
+}
+
+sub _make_iterator_for_template_allocations {
+    my $self = shift;
+    my $dbh = $self->_dbh_for_template();
+    return $self->_make_iterator_for_fetching_allocations($dbh);
+}
+
+sub _make_iterator_for_database_allocations {
+    my $self = shift;
+    my $dbh = $self->_dbh_for_database();
+    return $self->_make_iterator_for_fetching_allocations($dbh);
+}
+
+sub _make_check_for_production_allocation {
+    my $self = shift;
+
+    my $dbh = $self->_dbh_for_production_db();
+    Carp::croak("Can't connect to production DB: $DBI::errstr") unless $dbh;
+
+    my $sth = $dbh->prepare('SELECT id from disk.allocation where id = ?');
+    Carp::croak("Can't prepare select for allocations in production DB: $DBI::errstr") unless $sth;
+
+    return sub {
+        my $alloc_id = shift;
+        $sth->execute($alloc_id);
+        my $rv = $sth->fetchrow_arrayref() ? 1 : '';
+        $sth->finish;
+        return $rv;
+    };
+}
+
+my $sql_for_allocations = q(SELECT id, kilobytes_requested FROM disk.allocation ORDER BY id);
+sub _make_iterator_for_fetching_allocations {
+    my($self, $dbh) = @_;
+    my $sth = $dbh->prepare($sql_for_allocations);
+    $sth->execute();
+
+    return sub {
+        my @row = $sth->fetchrow_array;
+        return unless @row;
+        return Genome::Disk::StrippedDownAllocation->new(id => $row[0], kilobytes_requested => $row[1]);
+    };
+}
+
+sub disconnect_database_handles {
+    my $self = shift;
+    $_->disconnect foreach ( $self->_dbh_for_template, $self->_dbh_for_database);
+}
+
+
+sub _dbh_for_template {
+    my $self = shift;
+    return $self->_dbh_for('template');
+}
+
+sub _dbh_for_database {
+    my $self = shift;
+    return $self->_dbh_for('database');
+}
+
+sub _dbh_for {
+    my($self, $db_or_tmpl) = @_;
+
+    my $cache_key = "_cached_${db_or_tmpl}_handle";
+    unless ($self->{$cache_key}) {
+        my $db_name = $db_or_tmpl eq 'database'
+                        ? $self->database_name
+                        : $self->template_name;
+        my $db_host = $self->database_server;
+        my $db_port = $self->database_port;
+
+        $self->{$cache_key} = DBI->connect("dbi:Pg:dbname=${db_name};host=${db_host};port=${db_port}",
+                                           Genome::Config::get('ds_gmschema_login'),
+                                           Genome::Config::get('ds_gmschema_auth'),
+                                           { AutoCommit => 0, RaiseError => 1, PrintError => 0 });
+    }
+    return $self->{$cache_key};
+}
+
+my @ENV_VARS_FOR_TEST_DB = qw( XGENOME_DS_GMSCHEMA_LOGIN XGENOME_DS_GMSCHEMA_AUTH XGENOME_DS_GMSCHEMA_SERVER );
+sub _dbh_for_production_db {
+    my $self = shift;
+
+    delete local @ENV{ @ENV_VARS_FOR_TEST_DB };
```

Comment:
`Genome::Config::spec($key)` returns a `Genome::ConfigSpec` object which has `->env` to provide the environment variable name.  There is also `Genome::Config::default_value($key)` that will return the default value directly.